src/water.h

```diff
@@ -39,6 +39,7 @@ void MakeWaterKeepingClass(TileIndex tile, Owner o);
 void CheckForDockingTile(TileIndex t);
 
 bool RiverModifyDesertZone(TileIndex tile, void *data);
+void DoRiverModifyDesertZone(TileIndex tile, uint8 random_bits);
```

Comment:
The function does not only do that, but much more. What about: `MakeRiverAndMakeNearbyDesertTropical`? It's a longer name, but you immediately know what it's doing. Okay, this name is not perfect as it's not really "tropical" but the normal in the tropical map.
I'd maybe even rename `RiverModifyDesertZone` to `MakeDesertTropical`.